Packages/Blockchain/Sources/Solana/SolanaService.swift

```diff
@@ -402,7 +405,21 @@ extension SolanaService: ChainTokenable {
             type: .spl
         )
     }
-    
+
+    public func getTokenProgram(tokenId: String) async throws -> SolanaTokenProgram {
```

Comment:
should we check for token address instead? 

```
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "getAccountInfo",
  "params": [
    "2b1kV6DkPAnxd5ixfnxCpjxmKwqjjaYmCZfHsFu24GXo",
    {
      "encoding": "base64"
    }
  ]
}
```

In this case we don't need to get parsed data, simpler

Reply:
and we call this method `getProgramID` instead